File: src/lib/api/activity.ts
import { invoke } from "@tauri-apps/api/core";
import type { ActivityFilters, ActivitySnapshot, ActivityTrace } from "$lib/types";

export function listActivity(filters: ActivityFilters = {}): Promise<ActivitySnapshot> {
  return invoke<ActivitySnapshot>("list_activity", {
    workspace: filters.workspace ?? null,
    tool: filters.tool ?? null,
    status: filters.status ?? null,
    limit: filters.limit ?? 200,
  });
}

export function getActivity(traceId: string): Promise<ActivityTrace | null> {
  return invoke<ActivityTrace | null>("get_activity", { traceId });
}

export function clearActivity(): Promise<number> {
  return invoke<number>("clear_activity");
}

File: src-tauri/src/commands/activity.rs
use tauri::State;

use crate::activity::{ActivityQuery, ActivitySnapshot, ActivityTrace};
use crate::app_state::AppState;

#[tauri::command]
pub fn list_activity(
    state: State<'_, AppState>,
    workspace: Option<String>,
    tool: Option<String>,
    status: Option<String>,
    limit: Option<usize>,
) -> ActivitySnapshot {
    state.activity.snapshot(&ActivityQuery {
        workspace: workspace.unwrap_or_default(),
        tool: tool.unwrap_or_default(),
        status: status.unwrap_or_default(),
        limit: limit.unwrap_or(200),
    })
}

#[tauri::command]
pub fn get_activity(state: State<'_, AppState>, trace_id: String) -> Option<ActivityTrace> {
    state.activity.get(&trace_id)
}

#[tauri::command]
pub fn clear_activity(state: State<'_, AppState>) -> usize {
    state.activity.clear()
}
